### PR TITLE
The City Map

### PR DESCRIPTION
**Pull request by Kitsunemitsu**
## About The Pull Request
Adds in a new city map.
Adds in a new fixer office/syndicate creation system with capsules that spawn on the deisgnated landmark
Adds in a Hana Potential tester that gives you an average of your stats and a stationary version of it.
Adds in an item selling system that sells certain items on city maps.
Adds in various vendors for the city
Adds in a way for civilians and rats to increase stats
Adds in Hana, Blade Lineage, Ncorp, Thumb, Kurokumo and Index jobs
Adds in Thumb, Kurokumo and index armor.

Workshop related stuff
Adds in 4 new Workshop mods
Adds in 4 new workshop templates

Kinda Related: 
Fixes Hana PE sales to actually have the right name and removes the duplicates, this is kinda related in that this is for the backstreets
Throwing weapons are very common on city maps, this fixes issues regarding them ignoring defense.

## Why It's Good For The Game
tbd

## Changelog
:cl:
tbd
/:cl:

### CODE REVIEW

**Comment by LanceSmites328** [comment]
Armor plates are quite expensive for needing 3 of them to get any effect, 5 after that, and then another 7 after that.

**Comment by Kitsunemitsu** [comment]
Armor plates were removed due to me not liking htem, an armor workshop will show up later

**Comment by Kitsunemitsu** [comment]
So I fucked up my branch HARD.
I don't want to touch it for fear of undoing like 30 hrs of work.
Make a list of fixes for me to make and I'll add them over the course of a week.

**Comment by Kitsunemitsu** [comment]
This PR cannot be changed from this state, and should be merged with a list of changes for a future PR.

File: ModularTegustation/tegu_items/workshop/templates/basic.dm [hunk 1]
```diff
@@ -170,3 +188,30 @@ Katana - Use in hand to dash
...
+
+/obj/item/ego_weapon/template/club/attack(mob/living/target, mob/living/user)
+	if(!CanUseEgo(user))
+		return
+	. = ..()
+	var/atom/throw_target = get_edge_target_turf(target, user.dir)
+	if(!target.anchored)
+		var/whack_speed = (prob(60) ? 1 : 4)
+		target.throw_at(throw_target, rand(1, 2), whack_speed, user)
+
```
Comment (LanceSmites328) [inline ModularTegustation/tegu_items/workshop/templates/basic.dm:217]: Duplicate

File: ModularTegustation/tegu_items/workshop/templates/dice.dm [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	finishedicon = list("finisheddice")
+	finishedname = list("dice")
+	finisheddesc = "A finished dice, ready for use."
+	var/realforce	//Holds onto force
+
+/obj/item/ego_weapon/template/dice/attack(mob/living/target, mob/living/user)
+	realforce = force
+	force = rand(realforce*0.10, realforce)
+	..()
+	force = realforce
```
Comment (LanceSmites328) [inline ModularTegustation/tegu_items/workshop/templates/dice.dm:19]: forceholder exists for this purpose now, no?

File: code/modules/clothing/suits/ego_gear/non_abnormality/bladelineage.dm
Comment (LanceSmites328) [inline code/modules/clothing/suits/ego_gear/non_abnormality/bladelineage.dm:1]: I don't believe any of the changes in this file are accurate

File: code/modules/jobs/job_types/city/workshop.dm [hunk 3]
```diff
@@ -24,8 +24,8 @@ Workshop employee
 /datum/job/chef/after_spawn(mob/living/carbon/human/H, mob/M, latejoin = FALSE)
 	H.set_attribute_limit(0)
 	ADD_TRAIT(H, TRAIT_WORK_FORBIDDEN, JOB_TRAIT)
-	to_chat(M, "<span class='userdanger'>Your workshop is northwest, of the city. </span>")
-	to_chat(M, "<span class='userdanger'>Make weapons for the fixers who come asking in exchange for money. You cannot give away weapons for free </span>")
+	job_important = "Make weapons for the fixers who come asking in exchange for money. You cannot give away weapons for free."
+	job_notice = "Your workshop is northwest, of the city."
```
Comment (LanceSmites328) [inline code/modules/jobs/job_types/city/workshop.dm:28]: > Chef

File: code/modules/reagents/chemistry/reagents/withdrawal/generic_addictions.dm
Comment (LanceSmites328) [inline code/modules/reagents/chemistry/reagents/withdrawal/generic_addictions.dm:1]: these seem to mimic smoking pr, watch them for errors

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm
Comment (LanceSmites328) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm:1]: these seem to mimic smoking pr, watch them for errors

File: _maps/templates/fixer_office/workshop.dmm
Comment (LanceSmites328) [inline _maps/templates/fixer_office/workshop.dmm:1]: I'd like an ordeal extractor in this at some point.

**Review by LanceSmites328 — APPROVED** [review]
Consider this the requested change list.

File: code/modules/clothing/suits/ego_gear/non_abnormality/bladelineage.dm
Comment (Kitsunemitsu) [reply code/modules/clothing/suits/ego_gear/non_abnormality/bladelineage.dm:1]: They actually are. 
Cutthroats are the leaders according to Outis, who is the leader.
Project moon are just dumb fucking idiots and made the only 00 the murderhobo leader and all the others are 000s
Descriptions will be fixed in just a second.

image

File: code/modules/jobs/job_types/city/workshop.dm (hunk 3 above)
Comment (Kitsunemitsu) [reply code/modules/jobs/job_types/city/workshop.dm:28]: Nah, this job only exists on Ecorp

File: code/modules/reagents/chemistry/reagents/withdrawal/generic_addictions.dm
Comment (Kitsunemitsu) [reply code/modules/reagents/chemistry/reagents/withdrawal/generic_addictions.dm:1]: Weird.

File: ModularTegustation/tegu_items/workshop/templates/dice.dm (hunk 2 above)
Comment (Kitsunemitsu) [reply ModularTegustation/tegu_items/workshop/templates/dice.dm:19]: Fuck I didn't notice that existed.

File: code/modules/jobs/job_types/city/workshop.dm (hunk 3 above)
Comment (LanceSmites328) [reply code/modules/jobs/job_types/city/workshop.dm:28]: > Nah, this job only exists on Ecorp

Yes *but* this is in the file for workshop job

File: code/modules/clothing/suits/ego_gear/non_abnormality/bladelineage.dm
Comment (LanceSmites328) [reply code/modules/clothing/suits/ego_gear/non_abnormality/bladelineage.dm:1]: No I meant the description includes the % speed but the variable for it is incorrect

File: code/modules/jobs/job_types/city/workshop.dm (hunk 3 above)
Comment (Kitsunemitsu) [reply code/modules/jobs/job_types/city/workshop.dm:28]: OH GOD

File: code/modules/clothing/suits/ego_gear/non_abnormality/bladelineage.dm
Comment (Kitsunemitsu) [reply code/modules/clothing/suits/ego_gear/non_abnormality/bladelineage.dm:1]: Yep, that's been fixed, not pushed

File: ModularTegustation/tegu_items/workshop/templates/dice.dm [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+	armortype = RED_DAMAGE
+	finishedicon = list("finisheddice")
+	finishedname = list("dice")
+	finisheddesc = "A finished dice, ready for use."
+
+/obj/item/ego_weapon/template/dice/attack(mob/living/target, mob/living/user)
+	forceholder = force
+	force = rand(forceholder*0.10, forceholder)
+	..()
+	force = forceholder
```
Comment (LanceSmites328) [inline ModularTegustation/tegu_items/workshop/templates/dice.dm:18]: This method will not work due to it inheriting Template's attack, which sets forceholder equal to force. This means it goes:
Forceholder = force, force = rand, forceholder = force, force = forceholder, force = forceholder

**Review by LanceSmites328 — APPROVED** [review]

**Pull request merged**